Some updates; mostly issues with the ECS agent re-registering itself on reboot.

gswallow opened the pull request:
See http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_instances.html#container_instance_life_cycle for their description of a bug in the ECS agent < 1.8.1.

gswallow committed:
/var/run is tmpfs so when a minion reboots, the ECS agent re-registers as new.

Also, Amazon says there's a bug with registration that is fixed in v1.8.1 of their agent.

gswallow committed:
Updates.

phobologic commented on ansible/roles/docker/tasks/main.yml:
```diff
@@ -42,7 +42,7 @@
   tags: 
     - build_ami
-- apt: name=docker-engine=1.7.0-0~trusty
+- apt: name=docker-engine=1.9.1-0~trusty
```
We've ran into issues w/ 1.9 - honestly anything past 1.7 has led to issues with the logging infrastructure.  We're waiting for 1.11, which willhopefully fix some of those issues with the new logging drivers, and then we'll be releasing a new logging infrastructure along with it.

gswallow replied:
Docker 1.11 is out afaik.  I'm wondering if the round-robin DNS features in Docker 1.11 mean that ECS will schedule more than one web worker on a minion?

I figured that my update to docker engine 1.9.1 might raise questions.  I think we got rid of the cloudwatch/kinesis logging in lieu of shipping logs directly to sumologic.

phobologic replied:
Yep - just got released - we're hoping to get a chance to look at it soon.  Eric did a bunch of work testing out 1.10 with the new logging drivers, but the performance was poor (and the API logging performance fell apart as well - https://github.com/docker/docker/issues/20586).  I can't actually remember what issues we ran into with 1.9.

gswallow committed:
Use the devicemapper storage driver's direct-lvm mode.

Some docker stuff will still be stored in /var/lib/docker (resolv.conv for containers, etc.) but the storage driver will create miniature xfs volumes on raw storage.
You should make sure that you allocate a sufficiently large root ebs volume.

phobologic commented:
Hey @gswallow - so we're actually in the works of releasing a new version of the AMI & the stacker blueprints.  @mhahn is leading that effort, and it's going to include a _lot_ of new stuff, including docker 1.11.2, the latest ECS, etc.  You can find those efforts here: https://github.com/remind101/empire_ami/pull/26

Not sure if we should close this and potentially roll your changes up into the AMI that we're working on releasing, or what.  I actually don't have a strong handle, yet, on what the direct-lvm mode change will mean.

gswallow commented:
This particular issue, I think it's safe to close.  I was talking to you guys in the slack room today and I was asked by @ejholmes to submit an issue.

Since we are already using direct-lvm mode, and I had the changes to Ansible, I figured I would just submit a pull request.

phobologic commented:
Cool - thanks!

gswallow commented on ansible/roles/aws_ecs/tasks/main.yml:
```diff
@@ -1,5 +1,5 @@
 - include_vars: aws_ecs.yml
-- file: name=/var/run/ecs_agent mode=0750 owner=root group=root state=directory
+- file: name=/var/lib/ecs_agent mode=0750 owner=root group=root state=directory
```
This change should be considered.  /var/run gets wiped out on reboot, and the ECS agent registers itself as a new instance every time a minion is restarted.

mwildehahn replied:
verified this is happening

mwildehahn replied:
will include in the `empire-daemon` PR

gswallow commented:
Oh, damn.

There _is_ a change in there you should consider.  I added a comment to the
line that covers it.

On Wed, Jun 22, 2016 at 4:13 PM, Michael Barrett notifications@github.com
wrote:

> Cool - thanks!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/remind101/empire_ami/pull/20#issuecomment-227862633,
> or mute the thread
> https://github.com/notifications/unsubscribe/AEB7qDFT00XkSeZqNJDp7hh9ie_6jEiOks5qOZdjgaJpZM4IAbDb
> .
